af_ktls.c

```diff
@@ -984,10 +988,14 @@ static inline ssize_t tls_peek_data(struct tls_sock *tsk, unsigned flags)
 			goto peek_failure;
 		}
 
+		if (peeked_size < KTLS_MIN_RECORD_SIZE(tsk)) {
+			ret = -EBADF;
```

Comment:
>>  When peeked size is bigger than header: Only peek based on the header size
> I wanted to avoid the overhead of doing peek twice just to get actual records size. Now I am thinking whether we can actually avoid it.

Will not be easy. Note however, that the first peek will obtain only the TLS/DTLS header so it will not have any significant cost.